cluster/kubernetes/kubernetes.go

```diff
@@ -569,10 +569,21 @@ func (c *Cluster) ImagesToFetch() (imageCreds registry.ImageCreds) {
 				c.logger.Log("err", errors.Wrapf(err, "getting secret %q from namespace %q", secName.Name, service.Namespace))
 				continue
 			}
-			if sec.Type != v1.SecretTypeDockercfg {
+
+			var decoded []byte
+			var ok bool
+			// These differ in format; but, ParseCredentials will
+			// handle either.
+			switch api.SecretType(sec.Type) {
```

Comment:
NB this cast: the type of the field is v1.SecretType, but the value(s) we want to test against are type `api.SecretType` (both aliases of string). There is some k8s API versioning tomfoolery I'm missing here, since you can definitely have both kinds of secret.